### PR TITLE
Initial web app commit

### PR DESCRIPTION
**Pull request by FrankX0**
Initial graphical user interface for web control of the Pi word-clock.
Still work in progress.
To access: use "App" link on index page.

I welcome feedback!

### CODE REVIEW

**Comment by bk1285** [comment]
Hey @FrankX0,

that's a great starting point for a future webinterface! I did some first tests, finding the following points:

- the selection of plugins currently doesn't work.
- I will update the API to allow you to initialise brightness and color with the currently set values

I didn't have a closer look at the sources yet, but keep you updated.

Keep up the good work!

Best,
Bernd

**Comment by FrankX0** [comment]
Selection of plugins works again.

**Comment by bk1285** [comment]
Hey @FrankX0,

just found this page, which makes the experience very "app-like"... ;)

Best,
Bernd

https://developers.google.com/web/fundamentals/native-hardware/fullscreen/

File: wordclock_plugins/time_default/time_dutch.py [hunk 1]
```diff
@@ -9,11 +9,11 @@ class time_dutch:
     range(7,11) + range(40,44): VIJF OVER
     range(11,15) + range(40,44): TIEN OVER
     range(28,33) + range(40,44): KWART OVER
-    range(11,15) + range(18,22) + range(35,39): TIEN VOOR HALF
```
Comment (bk1285) [inline wordclock_plugins/time_default/time_dutch.py]: these fixes refer to this stancil layout? https://github.com/bk1285/rpi_wordclock/blob/master/wordclock_layouts/dutch_10x9/stancil_front.png

If not, I'd recommend a second version of time_dutch

File: wordclock_interfaces/templates/scripts/scripts.js [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+			this.$http.get('/api/plugin').
+			  then(this.successCallback2, this.errorCallback);
+			this.$http.get('/api/plugins').
+			  then(this.successCallback, this.errorCallback);
+		},
+		successCallback: function(response) {
+			this.apiData = response.data;
+			console.log('successCallback this.apiData:' , this.apiData);
+		},
+		successCallback2: function(response) {
```
Comment (bk1285) [inline wordclock_interfaces/templates/scripts/scripts.js]: Please rename to make the meaning a bit clearer.....

File: wordclock_plugins/time_default/time_dutch.py (hunk 1 above)
Comment (HeerNMeester) [reply wordclock_plugins/time_default/time_dutch.py]: There are two Dutch layouts an old 10x9 and a newer 11x10. I thought I committed a 11x10 stencil as well with my Dutch 11x10 code. But I don't see it in the Master. I think FrankXO uses the 11x10 as well. There is also a time_dutch109.py

Comment (FrankX0) [reply wordclock_plugins/time_default/time_dutch.py]: This is/was pull request #90 for Dutch 11x10.
It got lost again in the develop branch I guess.

File: wordclock_interfaces/templates/scripts/scripts.js (hunk 2 above)
Comment (FrankX0) [reply wordclock_interfaces/templates/scripts/scripts.js]: I will improve.

**Comment by FrankX0** [comment]
Looking at the fullscreen mode: works in Chrome on Windows but not (yet?) in Chrome on iOS 😢 
I would like to keep the PR open: I still plan to improve further.

File: wordclock_plugins/time_default/time_dutch.py (hunk 1 above)
Comment (bk1285) [reply wordclock_plugins/time_default/time_dutch.py]: Indeed, layout + code got somehow lost. I'll take care to restore them again (on develop first...)

File: wordclock_interfaces/templates/scripts/scripts.js [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+		updateBrightness: function(brightness) {
+			console.log('updateBrightness:' , brightness);
+			var xmlhttp = new XMLHttpRequest();
+			xmlhttp.open("POST", "/api/brightness");
+			xmlhttp.setRequestHeader("Content-Type", "application/json;charset=UTF-8");
+			xmlhttp.send(JSON.stringify({ "brightness": brightness }));
+		},
+		setColourWheel: function(wheel) {
+			var r,g,b,type;
+			[g,r,b] = hslToRgb(wheel.hue/360, wheel.saturation/100, wheel.luminosity/100);
```
Comment (bk1285) [inline wordclock_interfaces/templates/scripts/scripts.js]: Adopt to [r,g,b]
... results from fix for https://github.com/bk1285/rpi_wordclock/issues/99

Comment (FrankX0) [reply wordclock_interfaces/templates/scripts/scripts.js]: This was actually a temporary fix so I can at least set the right colour through the colour wheel.
The rest of the plugins still have red and green swapped.

File: wordclock_interfaces/templates/app.html [hunk 4]
```diff
@@ -0,0 +1,120 @@
+<!DOCTYPE html>
+<html>
+	<head>
+	<title>Word Clock</title>
+	<link rel='stylesheet prefetch' href='https://fonts.googleapis.com/css?family=Roboto:300,400,500,700|Material+Icons'>
+	<link rel='stylesheet prefetch' href='https://unpkg.com/vuetify@1.0.18/dist/vuetify.min.css'>
+	<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1, user-scalable=no, minimal-ui">
```
Comment (bk1285) [inline wordclock_interfaces/templates/app.html]: Concerning the full-screen: You can add here:
`    <meta name="mobile-web-app-capable" content="yes">`
`    <meta name="apple-mobile-web-app-capable" content="yes">`
... to have the page in full-screen on android + ios devices.

**Comment by FrankX0** [comment]
Superseded by #100

**Pull request closed without merging**